src/parser/core/tests/CombatLogParserIntegrationTest.js

```diff
@@ -0,0 +1,82 @@
+import { i18n } from 'interface/RootLocalizationProvider';
+
+import fs from 'fs';
+import zlib from 'zlib';
+
+/**
+ * Generates an integration test for a spec's CombatLogParser instance.
+ *
+ * The integration test applies the parser to a single fight that is
+ * downloaded and stored in the repository.
+ *
+ * `console.log` and `console.warn` are suppressed during the test, but
+ * `console.error` is not. This behavior can be disabled with the
+ * `suppressWarn` and `supressLog` parameters.
+ *
+ * To obtain the necessary JSON, load the report of interest and get the
+ * URLs for each call to the WoWAnalyzer API from the 'Network' section
+ * of the developer tools. Download the contents of each URL and store
+ * them somewhere. The full set of events should be gzipped (which
+ * reduces space usage from several MB to ~100kb).
+ *
+ * See the Brewmaster analyzer for a worked example.
+ *
+ * @param {object} parserClass - (uninstantiated) CombatLogParser subclass to test.
+ * @param {string} reportPath - Path to the report JSON.
+ * @param {string} combatantInfoPath - Path to the combatant info JSON.
+ * @param {string} eventPath - Path to the gzipped combat event JSON.
+ * @param {number} fightId - ID of the fight to parse.
+ * @param {number} playerId - ID of the player to render.
+ * @param {boolean} suppressWarn - Suppress `console.warn`
+ * @param {boolean} suppressLog - Suppress `console.log`
+ */
+export default function integrationTest(parserClass, reportPath, combatantInfoPath, eventPath, fightId, playerId, suppressWarn=true, suppressLog=true) {
+  return () => {
+    let report;
+    let combatantInfoEvents;
+    let events; 
+    beforeAll(() => {
+      report = JSON.parse(fs.readFileSync(reportPath).toString());
+      combatantInfoEvents = JSON.parse(fs.readFileSync(combatantInfoPath).toString()).events;
+      events = JSON.parse(zlib.unzipSync(fs.readFileSync(eventPath)).toString()).events;
+    });
+
+    const _console = {};
+    beforeEach(() => {
+      if(suppressWarn) {
+        _console.warn = console.warn;
+        console.warn = () => undefined;
+      }
+      if(suppressLog) {
+        _console.log = console.log;
+        console.log = () => undefined;
+      }
+    });
+
+    afterEach(() => {
+      if(suppressWarn) {
+        console.warn = _console.warn;
+      }
+      if(suppressLog) {
+        console.log = _console.log;
+      }
+    });
+
+    it('should parse the example report without crashing', () => {
+      const parser = new parserClass(
+        report, 
+        report.friendlies.find(({id}) => id === playerId),
+        report.fights.find(({id}) => id === fightId),
+        combatantInfoEvents
+      );
+      events.forEach(event => parser.getModule(EventEmitter).triggerEvent(event));
+      parser.finish();
+      const results = parser.generateResults({
+        i18n,
+        adjustForDowntime: false,
+      });
+      expect(results).toBeTruthy();
```

Comment:
I strongly believe you should make the results available for testing. This test isn't nearly as interesting as a test would be that validates if "statistic Y" has the value "1337%", which ensures the validation results (which should be fully consistent regardless of base changes) don't change.

Reply:
i agree. right now i haven't quite figured out a good way of generating the expected output to be stored in the repository. I've been considering generating & saving the results if they don't exist (and failing the test so that forgetting to commit the expected output fails travis), then checking against them if they do. so then adding consistency tests for a new module is as simple as adding a test like:

```js
describe("MyNewModule", () => {
  // other tests
  consistencyTest(CombatLogParser, MyNewModule, "log-identifier");
});
```
and then `git add <expected output file>`.

I would like to strike a balance between automation and brittle testing, because doing strict equality will likely produce tests that fail often for irrelevant changes (e.g. an invisible change to the html produced by the `StatisticBox` shouldn't fail the test). I haven't done much testing yet but I'm thinking of using a list of props to check for consistency so that module authors can pick out the things they believe should be tested.

Reply:
just do a snapshot test

Reply:
I didn't realize those existed. Looks good :+1:

Reply:
How about making this a snapshot test?

Reply:
So my reason for leaving this as a simple crash test is twofold:

1. Blindly testing against the total serialized output means that the answer to the question "should this test have broken?" is almost always yes. Virtually every core or spec-local change will change *some* output by definition, so the having one big snapshot test only really checks if another spec's changes changed this one.

2. Testing against serialized output of each individual module in an automated fashion is much more complex. I think it is probably possible to do, and it'd likely be value in having a loop over the (spec-local) modules that effectively does `expect(module.statistic()).toMatchSnapshot()` on each instead of requiring it be done individually. This covers 99% of our problem cases in a manner that avoids the "one big snapshot" issue.

As a matter of fact I have talked myself into the latter solution 🤔 im not implementing it tonight, but i think i know how. i'll follow up with you tomorrow or thursday about it.

Reply:
I would like to be able to see the impact of core changes.

Reply:
Maybe we need to set up optional tests with full snapshots, or a bot doing it in addition to the travis tests